utils/link_utils.js

```diff
@@ -0,0 +1,67 @@
+const axios = require('axios');
+const { ytApiId } = require('../config.json');
+
+// regex list with all compatible
+const sourceRegexList = {
+    YouTube: /^(?:https?:\/\/)?(?:www\.)?(?:youtube\.com\/watch\?v=|youtu\.be\/)([a-zA-Z0-9_-]+)/,
+    YouTubePlaylist: /^(?:https?:\/\/)?(?:www\.)?youtube\.com\/playlist\?list=([a-zA-Z0-9_-]+)/,
+};
+
+// matches url against regex
+function identifyUrlType(url) {
+    for (const source in sourceRegexList) {
+        const regex = sourceRegexList[source];
+        const match = url.match(regex);
+        if (match) {
+            return { source, id: match[1] };
+        }
+    }
+    return { source: null, id: null };
+}
+
+// converts yt playlist to a list of (YouTube, <url>)
+async function getYtItemsFromPlaylist(playlistId) {
+    const url = `https://www.googleapis.com/youtube/v3/playlistItems?part=snippet&playlistId=${playlistId}&maxResults=100&key=${ytApiId}`;
+    try {
+        const response = await axios.get(url);
+        const videoIds = response.data.items.map(item => ({ source: 'YouTube', id: item.snippet.resourceId.videoId }));
```

Comment:
currently I hardcode source to the same string as the key in the dictionary with with regex.
I am not sure if we should use numbers (less clarity, less memory usage), or if there is another way I can handle this.

(Context: we need to change a YouTube playlist to a list of YouTube links. Given that we might also offer support for other platforms, we need to tag the links appropriately).

Reply:
same as below